### PR TITLE
[move] Create separate burn account

### PR DESCRIPTION
**Pull request by andll**
This diff creates separate account for burning.

This is needed so we can rotate key of this account to HSM key.

We can not rotate main association key, because if we do everything else(like minting from faucet) will stop working.

### CODE REVIEW

**Comment by sblackshear** [comment]
This looks great to me!

`0xA550C18` still has a `BurnCapability` given to it by `register_currency`--we might want to get rid of that at some point. However, that will break all of the tests for burning under `move-lang/tests/functional/libra` because they all send burn transactions from `0xA550C18`.

The long-term fix is probably:
- Add a functional test directive for sending a tx from `0xD1E` (CC @vgao1996) and use that in the burn tests
- Remove the `BurnCapability` from `0xA550C18`

File: language/tools/vm-genesis/src/lib.rs [hunk 1]
```diff
@@ -586,6 +660,19 @@ fn create_and_initialize_main_accounts(
         )
         .expect("Failure rotating association key");
 
+    let genesis_auth_key = AuthenticationKey::ed25519(public_key).to_vec();
```
[the remark this answers is not in the document]
Comment (andll) [reply language/tools/vm-genesis/src/lib.rs:663]: I think for now we are going to have same as this is simplest way :)
In production we are going to put here public key from vault

File: language/stdlib/modules/libra.move [hunk 2]
```diff
@@ -147,6 +147,10 @@ module Libra {
         BurnCapability<CoinType> { }
     }
 
+    public fun grant_burn_capability_for_sender<CoinType>() {
```
[the remark this answers is not in the document]
Comment (andll) [reply language/stdlib/modules/libra.move:150]: Added

**Pull request closed without merging**